ci_framework/roles/reproducer/tasks/ci_job.yml

```diff
@@ -177,29 +183,38 @@
         dest: "src/github.com/openstack-k8s-operators/ci-framework/pre-ci-play.yml"
         src: "pre-ci-play.yml"
 
-    - name: Check for ansible log file and rotate it
+    - name: Push zuul-params.yml to expected location
+      tags:
+        - bootstrap
       block:
-        - name: Check ansible.log
-          register: ansible_log
-          ansible.builtin.stat:
-            path: ansible.log
-
-        - name: Rotate log if present
-          when:
-            - ansible_log.stat.exists
-          ansible.builtin.shell:
-            cmd: mv ansible.log "ansible.log-$(date +%s)"
-
-        - name: Check pre-ci log
-          register: pre_ci_log
-          ansible.builtin.stat:
-            path: ansible-pre-ci.log
-
-        - name: Rotate pre-ci log if present
-          when:
-            - pre_ci_log.stat.exists
-          ansible.builtin.shell:
-            cmd: mv ansible-pre-ci.log "ansible-pre-ci.log-$(date +%s)"
+        - name: Get content of zuul-params.yml
+          register: zuul_params
+          ansible.builtin.slurp:
+            path: "{{ job_id }}-params/zuul-params.yml"
+
+        - name: Push extracted content
+          vars:
+            zuul_params_filtered: >-
+              {{
+                (zuul_params['content'] | b64decode | from_yaml) |
+                dict2items |
+                rejectattr('key', 'equalto', 'cifmw_operator_build_output') |
+                rejectattr('key', 'equalto', 'content_provider_registry_ip:') |
+                items2dict
+              }}
+          ansible.builtin.copy:
+            dest: "ci-framework-data/artifacts/parameters/zuul-params.yml"
+            content: "{{ zuul_params_filtered | to_nice_yaml }}"
+
+    - name: Check for ansible logs file and rotate it
```

Comment:
this is a nice optimization using loop and a rotate_log task file 

as a general comment this review could have been split into smaller commits (eg bits like this) for better separation and review

Reply:
Yeah... maybe. Well, I had to work my way through too many things at the same time, and I got pretty annoyed with the logs rotation and the copy-pasting thing :).